### ISSUE TITLE
[BUG] Possible fix for flickering glass? 

### ISSUE THREAD

**Issue by CoasterBlock** [tsunamayo/Starship-EVO#5301]
### Short Description of the issue:

**I may have found the reason, or a possible reason for the flickering glass**

So for understand, this is what I mean when I say "texture channels":

**Glossy: Metallness/shine of a brick. 
Color: The RGB value used to paint
Alpha texture: The value used to define transparency.** 

**Glass-brick: The regular glass bricks
Framed-glass: The framed glass that comes in clear, and two frosted variants.**

### Observations: Glass flicker occurs in a very specific way:

My thinking behind that is watching the way the glass flickers, because note: there are the glass bricks that flicker, AND the framed glass. The framed glass doesnt flicker at a distance, and it interestingly lacks the color-layer on the glass texture. It only has the glossy, and the alpha texture. While the glass-bricks have all three. You can change the material (glossy texture) and the color (color texture) on the brick. The framed glass can also be changed on its frame, the glass itself stays the same.

therefore: it has to be something about the specific way color, glossy, and alpha work on those bricks. Because the framed glass ALSO flickers when viewing regular glass bricks and the framed glass also flickers when viewed through regular glass bricks.

### Observation;

framed + framed = no flicker 

glass-brick + glass-brick= flicker

glass-brick + framed glass = flicker.

### ADDENDUM: 

Looking closely at the flickering, it seems to be specifially the metall-texture/glossy that flickers. If it where simple z-fighting all channels would seem to flicker, and they wouldnt flicker at a distance so it must be specifically the glass bricks that do something strange using the texture channels

Tldr: Looking at the glass in game, it seems to be specifically the channel that can be painted using the material paints (Metallness/Glossy), because that one seems to be flickering when viewing glass through other glass-bricks. 

These are the observations I made while using large glass-panes of differing types. I hope this will help with bug hunting.

Picture for clarifictation:

![Glass Flicker](https://github.com/tsunamayo/Starship-EVO/assets/126499309/a33d2c2a-20f4-4e2d-be53-995704bcaaef)

**Comment by tsunamayo** [comment]
@CoasterBlock Hi, Well, I know exactly why glass flicker since before I even saw the bug, because it is a feature that is not yet wadded to the rendering engine: It is a rendering order issue, you should draw transparent surface back to front and I dont unfortunately.
Thanks

**Comment by CoasterBlock** [comment]
Ah very interesting!
